crates/consensus/src/header.rs

```diff
@@ -12,6 +12,35 @@ use alloy_primitives::{
 use alloy_rlp::{length_of_length, BufMut, Decodable, Encodable};
 use core::mem;
 
+/// Consensus Errors
+#[derive(Debug, PartialEq, Eq, Clone, derive_more::Display, derive_more::Error)]
+pub enum ConsensusError {
+    /// Error when blob gas used is missing.
+    #[display("missing blob gas used")]
+    BlobGasUsedMissing,
+
+    /// Error when excess blob gas is missing.
+    #[display("missing excess blob gas")]
+    ExcessBlobGasMissing,
+
+    /// Error when there is an invalid excess blob gas.
+    #[display(
+        "invalid excess blob gas, got: {got}, expected: {expected}; \
+            parent excess blob gas: {parent_excess_blob_gas}, \
+            parent blob gas used: {parent_blob_gas_used}"
+    )]
+    ExcessBlobGasDiff {
+        /// Expected excess blob gas.
+        expected: u64,
+        /// Actual excess blob gas.
+        got: u64,
```

Comment:
I've seen several places in alloy where the got/expected error pattern is present, maybe it would be good to move the reth primitive

https://github.com/paradigmxyz/reth/blob/ab07fcfb113cb0b579e1f9f55a5dd9511b576687/crates/primitives-traits/src/error.rs#L9

into alloy-primitives to be able to use it everywhere